lib/Doctrine/DBAL/Connection.php

```diff
@@ -189,7 +190,7 @@ public function __construct(
         $this->params  = $params;
 
         if (isset($params['pdo'])) {
-            $this->_conn       = $params['pdo'];
+            $this->_conn       = WrappedPDOConnection::fromInstance($params['pdo']);
```

Comment:
It still would be a breaking change since it's no longer a PDO instance (won't match a type hint, can't call PDO-specific methods, etc).

Reply:
Not according to our docblocks...

Reply:
We never expect to handle raw PDO connection in the internals

Reply:
True. I didn't notice it was about the internal API. But still, even if we can afford introducing this class, I don't think we should. Having two PDO connections which share most of the code is not a proper solution. I'd like to propose #3549 instead.

Reply:
@morozov I'm fine with not adding this adapter :smile: I'd suggest to port the tests to your PR, just to ensure that the basic stuff works fine.